src/Ar/VarTools/varGetValue.c

```diff
@@ -193,9 +194,9 @@ unsigned short varGetValue(unsigned long pVariable)
 	
 		case VAR_TYPE_LREAL:
 		
-			valueReal=	(REAL)*(LREAL*)(ipVariable->address);
+			valueLreal=	*(LREAL*)(ipVariable->address);
 		
-			brsftoa( valueReal, (UDINT)ipVariable->value );
+			stringdtoa(valueLreal, ipVariable->value, 0, sizeof(ipVariable->value));
```

Comment:
What is the interpretation of ndigits being 0 for this new stringdtoa function?

Reply:
In StringExt you can check out the gdtoa README which goes into detail, but essentially:
> do a "%g" style conversion either to a specified number of decimal
places (if their ndig argument is positive), or to the shortest
decimal string that rounds to the given binary floating-point value
(if ndig <= 0)